### Membuat DataFrame

In [1]:
import pandas as pd
import numpy as np

In [3]:
df = pd.DataFrame({"Transaction_Amount":[100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65]
                  })
df

Transaction_Amount
0                  100
1                  150
2                   50
3                  100
4                  130
5                  120
6                  100
7                   85
8                   70
9                  150
10                 150
11                 120
12                  50
13                 100
14                 100
15                 140
16                  90
17                 150
18                  50
19                  90
20                 120
21                 100
22                 110
23                  75
24                  65

### Analisis Measure of Central Tendency

In [9]:
Transaction_Amount = np.array([100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65])
Transaction_Amount

array([100, 150,  50, 100, 130, 120, 100,  85,  70, 150, 150, 120,  50,
       100, 100, 140,  90, 150,  50,  90, 120, 100, 110,  75,  65])

In [11]:
import numpy as np
import statistics
print(statistics.mean(Transaction_Amount))
print(statistics.median(Transaction_Amount))
print(statistics.multimode(Transaction_Amount))

102
100
[100]


### Analisis Measure of Variability

In [7]:
Transaction_Amount = np.array([100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65])
Transaction_Amount

array([100, 150,  50, 100, 130, 120, 100,  85,  70, 150, 150, 120,  50,
       100, 100, 140,  90, 150,  50,  90, 120, 100, 110,  75,  65])

In [9]:
import numpy as np
print(np.var(Transaction_Amount))
print(np.std(Transaction_Amount))
print(np.min(Transaction_Amount))
print(np.max(Transaction_Amount))
print(np.max(Transaction_Amount) - np.min(Transaction_Amount))
print(np.quantile(Transaction_Amount,0.25))
print(np.quantile(Transaction_Amount,0.5))
print(np.quantile(Transaction_Amount,0.75))

972.2399999999999
31.18076330047101
50
150
100
85.0
100.0
120.0


### Menentukan Hipoteis

Hipotesis0 : Penjualan rata-rata perusahaan X = 100

Hipotesis1 : Penjualan rata-rata perusahaan X > 100

#### Melakukan T-test pada data rata-rata jumlah transaksi dengan alpha=5%

In [21]:
import scipy.stats as stats

In [17]:
len(Transaction_Amount)

25

#### Menghitung t statistics dan P-value 
Menggunakan One Sample T-test

In [23]:
stats.ttest_1samp(a=Transaction_Amount, popmean=100)

TtestResult(statistic=0.4085001556802841, pvalue=0.6865284813438117, df=24)

Kesimpulan alpha dan P-value
Nilai P-value lebih besar dari alpha (5%), artinya kita gagal menolak H0
Kesimpulan : Training pada pekerja sales tidak efektif, rata-rata jumlah taransaksi masih 100

Kesimpulan t statistics dan critical region
Menentukan critical region

In [27]:
#H1 = larger (right side)
stats.t.ppf(1-0.05,24)

1.7108820799094275

Critical region : t > 1.71
Dengan hasil t statistics tidak jatuh di dalam critical region, maka kita gagal menolak H0
Kesimpulan : Training pada pekerja sales tidak efektif, rata-rata jumlah taransaksi masih 100